src/scope-hoisting/hoist.js

```diff
@@ -121,7 +121,7 @@ module.exports = {
         // Rename each binding in the top-level scope to something unique.
         for (let name in scope.bindings) {
           if (!name.startsWith('$' + t.toIdentifier(asset.id))) {
-            let newName = getName(asset, 'var', name);
+            let newName = getName(asset, 'var', name, '$');
```

Comment:
I don't think this should clash with anything. This would generate e.g. `$id$var$exports` vs our internal which is just `$id$exports`. Seems like the problem might be something else though I'm not sure why this would fix it...